### PR TITLE
5493 Enhance meandice

### LINKED ISSUE #5493: Implement Ensemble Dice
Reference:
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6454006/pdf/fbioe-07-00053.pdf

This metric is helpful for hovernet pipeline

cc @KumoLiu @drbeh

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Signed-off-by: Yiheng Wang <vennw@nvidia.com>

Fixes #5493 .

### Description

A few sentences describing the changes proposed in this pull request.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by yiheng-wang-nv** [comment]
Hi @wyli , @KumoLiu @Nic-Ma , may need your help to have a quick a look at this draft PR, I'm not confident with the naming of the changes, thanks!

File: monai/metrics/meandice.py [hunk 1]
```diff
@@ -44,6 +43,11 @@ class DiceMetric(CumulativeIterationMetric):
         ignore_empty: whether to ignore empty ground truth cases during calculation.
             If `True`, NaN value will be set for empty ground truth cases.
             If `False`, 1 will be set if the predictions of empty ground truth cases are also empty.
+        compute_sample: if ``True``, each sample's dice metric will be computed before reduction.
```
Comment (Nic-Ma) [inline monai/metrics/meandice.py:46]: Maybe call it `sample_wise`? @wyli What do you think?

Thanks.

Comment (wyli) [reply monai/metrics/meandice.py:46]: perhaps we just modify the existing `reduction` to support this sample mode? also for the typo, `dice` should be changed to `Dice`

Comment (drbeh) [reply monai/metrics/meandice.py:46]: I agree with @wyli on just modifying `reduction` method and having something like `reduction_sample` and use the `filename` attribute to accumulate the metrics.

**Review by myron — CHANGES_REQUESTED** [review]
This PR adds extra code to compute_dice() and DiceMetric to satisfy an edge case of non-standard metric calculation. 

If want to suggest to create a new class and call it something else , instead of adding more options and crowding the code of  compute_dice and DiceMetric

**Comment by yiheng-wang-nv** [comment]
Hi @myron @wyli @drbeh @Nic-Ma , thanks for the comments, I will close this PR and submit a new one later. I will create a new sub class rather than change the diceMetric

**Pull request closed without merging**